backend/src/test/java/com/example/backend/service/TripServiceTest.java

```diff
@@ -130,5 +131,91 @@ void deleteById_ThrowsTripNotRegisteredException_WhenNotRegistered() {
         Assertions.assertThrows(TripNotRegisteredException.class, () -> tripService.deleteById("abc1"));
     }
 
+    @Test
+    void update_updatesTrip_whenExists() throws TripNotRegisteredException {
+        // given
+        Trip trip = new Trip("some-id", "some trip", new ArrayList<>());
+
+        TripRepo tripRepo = mock(TripRepo.class);
+        when(tripRepo.existsById(any())).thenReturn(true);
+        when(tripRepo.save(any())).then(returnsFirstArg());
+
+        IdGenerator idGenerator = mock(IdGenerator.class);
+        when(idGenerator.generateRandomId()).thenReturn("new-location-id");
+
+        // when
+        TripService sut = new TripService(tripRepo, idGenerator);
+        Trip actual = sut.update(trip.getId(), trip);
+
+        // then
+        assertEquals(actual, trip);
+
+        verify(tripRepo).save(any());
+    }
+
+    @Test
+    void update_ensuresThatIdForNewLocationsIsGenerated() throws TripNotRegisteredException {
+        // given
+        Trip trip = new Trip("some-id", "some trip", new ArrayList<>(List.of(
+                new Location("Kölner Dom", 50.941386546092225, 6.958270670147375),
+                new Location("existing-location-id", "Planten un Blomen", 53.5625456617408, 9.98188182570993)
+        )));
+
+        TripRepo tripRepo = mock(TripRepo.class);
+        when(tripRepo.existsById(any())).thenReturn(true);
+        when(tripRepo.save(any())).then(returnsFirstArg());
+
+        IdGenerator idGenerator = mock(IdGenerator.class);
+        when(idGenerator.generateRandomId()).thenReturn("new-location-id");
+
+        List<String> expectedIds = List.of("new-location-id", "existing-location-id");
+
+        // when
+        TripService sut = new TripService(tripRepo, idGenerator);
+        Trip actual = sut.update(trip.getId(), trip);
+
+        // then
+        List<String> actualIds = actual.getLocations().stream().map(Location::getId).toList();
+        assertEquals(expectedIds, actualIds);
+
+        verify(idGenerator, times(1)).generateRandomId();
```

Comment:
That second argument is a really cool feature; being able to also verify how often you actually access a particular class is really quite handy, isn't it